cpp/include/cudf/io/parquet.hpp

```diff
@@ -859,82 +836,139 @@ class chunked_parquet_writer_options_builder {
+  parquet_chunked_writer() = default;
 
-/**
- * @brief Write a single table as a subtable of a larger logical parquet file/table.
- *
- * All tables passed into multiple calls of this function must contain the same # of columns and
- * have columns of the same type.
- *
- * @param[in] table The table data to be written.
- * @param[in] state Opaque state information about the writer process. Must be the same pointer
- * returned from write_parquet_chunked_begin().
- * @param[in] int96_timestamps Write out timestamps as INT96 type
- */
-void write_parquet_chunked(table_view const& table, std::shared_ptr<pq_chunked_state> state);
+  /**
+   * @brief Constructor with chunked writer options
+   *
+   * @param[in] op options used to write table
+   * @param[in] mr Device memory resource to use for device memory allocation
+   */
+  parquet_chunked_writer(
+    chunked_parquet_writer_options const& op,
+    rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
-/**
- * @brief Finish writing a chunked/stream parquet file.
- *
- * @param[in] state Opaque state information about the writer process. Must be the same pointer
- * returned from write_parquet_chunked_begin().
- * @param[in] return_filemetadata If true, return the raw file metadata.
- * @param[in] column_chunks_file_path Column chunks file path to be set in the raw output metadata.
- *
- * @return A blob that contains the file metadata (parquet FileMetadata thrift message) if
- *         requested in parquet_writer_options (empty blob otherwise).
- */
-std::unique_ptr<std::vector<uint8_t>> write_parquet_chunked_end(
-  std::shared_ptr<pq_chunked_state>& state,
-  bool return_filemetadata                   = false,
-  const std::string& column_chunks_file_path = "");
+  /**
+   * @brief Moves writer unique pointer to object
+   *
+   * @param[in] rhs writer class that needs to be moved
+   */
+  parquet_chunked_writer& operator=(parquet_chunked_writer&& rhs);
 
-/**
- * @brief Merges multiple raw metadata blobs that were previously created by write_parquet
- * into a single metadata blob
- *
- * @ingroup io_writers
- *
- * @param[in] metadata_list List of input file metadata
- * @return A parquet-compatible blob that contains the data for all rowgroups in the list
- */
-std::unique_ptr<std::vector<uint8_t>> merge_rowgroup_metadata(
-  const std::vector<std::unique_ptr<std::vector<uint8_t>>>& metadata_list);
+  /**
+   * @brief Writes table to output.
+   *
+   * @param[in] table Table that needs to be written
+   */
+  void write(table_view const& table);
+
+  /**
+   * @brief Finishes the chunked/streamed write process.
+   *
+   * @param[in] return_filemetadata If true, return the raw parquet file metadata
+   * @param[in] column_chunks_file_path Column chunks file path to be set in the raw output metadata
+   * @return unique_ptr to FileMetadata thrift message if requested
+   */
+  std::unique_ptr<std::vector<uint8_t>> write_end(bool return_filemetadata = false,
```

Comment:
Would be nice to have a destructor that does the same as `write_end`, so that users don't need to call `write_end` to have valid file written out once the writer goes out of scope.

Reply:
but we need the return value sometimes.

Reply:
we don't return anything by default, right?
I see a lot of tests where the return value is discarded, so I assume that many users don't really have the need to call write_end.

Reply:
Added support with destructor to write.